Sources/Persistence/ServerSyncJobStore.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+struct ServerSyncJobStore {
+
+    private var jobStoreArray: [ServerSyncJob] = []
+    private let syncJobStoreQueue = DispatchQueue(label: "syncJobStoreQueue")
+
+    init() {
+        self.jobStoreArray = self.loadOperations()
+    }
+
+    private func loadOperations() -> [ServerSyncJob] {
+        guard let operations = NSData(contentsOfFile: "syncJobStoreQueue") else {
+            return []
+        }
+
+        let jsonDecoder = JSONDecoder()
+        guard let operationsArray = try? jsonDecoder.decode([ServerSyncJob].self, from: (operations as Data)) else {
+            return []
+        }
+
+        return operationsArray
+    }
+
+    var isEmpty: Bool {
+        get {
+            return syncJobStoreQueue.sync {
+                return self.jobStoreArray.isEmpty
+            }
+        }
+    }
+
+    var first: ServerSyncJob? {
+        get {
+            return syncJobStoreQueue.sync {
+                return jobStoreArray.first
+            }
+        }
+    }
+
+    func toList() -> [ServerSyncJob] {
+        return syncJobStoreQueue.sync {
+            return self.jobStoreArray
+        }
+    }
+
+    mutating func append(_ job: ServerSyncJob) {
+        syncJobStoreQueue.sync {
+            self.jobStoreArray.append(job)
+
+            let jsonEncoder = JSONEncoder()
+            let data = try! jsonEncoder.encode(jobStoreArray)
+            try! (data as NSData).write(toFile: "syncJobStoreQueue", options: .atomic)
```

Comment:
Can you make the final name a constant? It's used 3 times here... It can be at the top of the file.